spec/support/macros.rb

```diff
@@ -1,11 +1,20 @@
 def set_current_user(user=nil)
-  session[:user_id] = (user || Fabricate(:user)).id 
+  session[:user_id] = (user || Fabricate(:user)).id
+end
+
+def set_current_admin(admin=nil)
+  session[:user_id] = (admin || Fabricate(:admin)).id
 end
 
 def sign_in(a_user=nil)
   user = a_user || Fabricate(:user)
```

Comment:
You should refactor this:

```ruby
 def sign_in(user = nil)
   user ||= Fabricate(:user)
```

Use conditional assignment / memoization skill.